### PR TITLE
Allow meta leader stepdown.

### PR DESCRIPTION
**Pull request by derekcollison**
Also easier way to put authorization on system account and still use $G.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by matthiashanel — CHANGES_REQUESTED** [review]
minor error checking and assumption behind jsLeaderStepDownRequest

File: server/jetstream_api.go [hunk 1]
```diff
@@ -1657,6 +1673,41 @@ func (s *Server) jsStreamRemovePeerRequest(sub *subscription, c *client, subject
 	s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(resp))
 }
 
+// Request to have the meta leader stepdown.
+// These will only be received the the meta leaders, so less checking needed.
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/jetstream_api.go:1677]: I double check if we are still the leader.

File: server/jetstream_api.go [hunk 2]
```diff
@@ -1657,6 +1673,41 @@ func (s *Server) jsStreamRemovePeerRequest(sub *subscription, c *client, subject
...
+	js.mu.RLock()
+	isLeader := cc.isLeader()
+	js.mu.RUnlock()
+
+	if !isLeader {
+		return
+	}
+
+	// Call actual stepdown.
+	cc.meta.StepDown()
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/jetstream_api.go:1704]: I could make that change.

File: server/server.go [hunk 3]
```diff
@@ -697,6 +704,22 @@ func (s *Server) configureAccounts() error {
 		if err != nil {
 			return fmt.Errorf("error resolving system account: %v", err)
 		}
+
+		// If we have defined a system account here check to see if its just us and the $G account.
+		// We would do this to add user/pass to the system account. If this is the case add in
+		// no-auth-user for $G.
+		if numAccounts == 2 {
```
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/server.go]: Yes did so for other reasons (broken test).

File: server/jetstream_api.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (derekcollison) [reply server/jetstream_api.go]: ok done.

File: server/jetstream_api.go [hunk 4]
```diff
@@ -1657,6 +1673,45 @@ func (s *Server) jsStreamRemovePeerRequest(sub *subscription, c *client, subject
...
+	js.mu.RLock()
+	isLeader := cc.isLeader()
+	js.mu.RUnlock()
+
+	if !isLeader {
+		return
+	}
+
+	// Call actual stepdown.
+	err := cc.meta.StepDown()
```
Comment (matthiashanel) [inline server/jetstream_api.go:1704]: err exists already...

File: server/jetstream_api.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiashanel) [reply server/jetstream_api.go:1677]: Maybe modify the comment?

**Review by matthiashanel — CHANGES_REQUESTED** [review]

File: server/jetstream_api.go (hunk 4 above)
Comment (derekcollison) [reply server/jetstream_api.go]: fixed

**Review by matthiashanel — APPROVED** [review]
LGTM

File: server/reload.go [hunk 5]
```diff
@@ -1063,6 +1063,23 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
...
+						return nil, fmt.Errorf("config reload not supported for %s: old=%v, new=%v",
+							field.Name, oldValue, newValue)
+					}
+				}
+			} else {
+				return nil, fmt.Errorf("config reload not supported for %s: old=%v, new=%v",
+					field.Name, oldValue, newValue)
+			}
+		case "systemaccount":
+			if oldValue != DEFAULT_SYSTEM_ACCOUNT || newValue != _EMPTY_ {
```
Comment (matthiashanel) [inline server/reload.go:1079]: I assume we won't get here if oldValue == new Value.

**Review by matthiashanel — APPROVED** [review]
LGTM

File: server/reload.go (hunk 5 above)
Comment (derekcollison) [reply server/reload.go:1079]: For this code no..

**Pull request merged**